main.js

```diff
@@ -0,0 +1,149 @@
+  defaults: function () {
+    return {
+      videoId: "",
+      videoThumbnamil: "",
+      videoTitle: "",
+      videoDesc: "",
+    };
+  },
+});
+
+// Videos collection - add API fetch here
+var VideosCollection = Backbone.Collection.extend({ 
+  url: function (topic) {
+    console.log(topic);
+    if (!topic) {
+      return "https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=5&q=cercle&type=video&videoEmbeddable=true&key=AIzaSyCjo4u-wcr_ExFNPxiYlWZP3LLr-ythijE";
+    } else {
+      return (
+        "https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=5&q=" +
+        topic +
+        "cercle&type=video&videoEmbeddable=true&key=AIzaSyCjo4u-wcr_ExFNPxiYlWZP3LLr-ythijE"
+      );
+    }
+  },
+
+  model: VideoModel,
+
+  initialize: function () {
+    this.on("add", function (model) {
+      model.fetch();
+    });
+  },
+  
+  parse: function (response) {
+    return response.items.map(function (x) {
+      return {
+        videoId: x.id.videoId,
+        videoThumbnail: x.snippet.thumbnails.default.url,
+        videoTitle: x.snippet.title,
+        videoDesc: x.snippet.description,
+      }
+    })
+  }
+});
+
+var AppModel = Backbone.Model.extend({
+  defaults: function () {
+    return {
+      videos: new VideosCollection(),
+    };
+  }
+});
+
+var VideoView = Backbone.View.extend({
+  className: 'video',  // to reference later
+
+  template: Handlebars.compile($('#five-display-template').html()),
+
+  render: function () {
+    this.$el.html(this.template(this.model.toJSON()));
+
+    return this;
+  },
+});
+
+var MainVideoView = Backbone.View.extend({
+  className: 'mainVideo',
+
+  template: Handlebars.compile($('#main-display-template').html()),
+
+  render: function () {
+    this.$el.html(this.template(this.model.toJSON()));
+
+    return this;
+  }
+})
+
+var AppView = Backbone.View.extend({
+  el: $("body"),
+
+  events: {
+    "click .search": "newSearch", 
+    "click .click-here": "newMain"
+  },
+  
+  initialize: function () {
+    this.$search = this.$('#search');
+    this.listenTo(this.model.get('videos'), 'change', this.renderVideo);
+    this.listenTo(this.model.get("videos"), "change", this.renderMainVideo);
+
+    this.$mainspot = this.$('.five-spot');
+    this.listenTo(this.model.get('videos'), 'reset', this.renderVideos);
+    this.listenTo(this.model.get("videos"), "reset", this.renderMainVideo);
+    
+    this.renderVideos;
+  },
+
+  newSearch: function () {
+    var inputTopic = this.$('.topic').val();
+
+    this.model.set({topic: inputTopic});
+    // listen to change of model should notice new url and run render functions
+  },
+
+  renderVideo: function (model) {
+    var videoView = new VideoView({ model: model });
+
+    this.$('.five-display').append(videoView.render().el);
+  },
+
+  renderVideos: function () {
+    this.model.get('videos').each(function (v) {
+      this.renderVideo(v);
+    }, this);
+  },
+
+  renderMainVideo: function (model, vidId) {
```

Comment:
The reason this function is overcomplicated is b/c you violated the rules of data-flow. You can also tell that theres a problem if your views are bloated and your models/collections are very small. It means the view is doing more work than it should.

`newMain` should merely update the `AppModel` with a new `current_video`. Then the `AppView` can listen for a change on `current_video` and then call `renderMainVideo`. 

Then you wouldn't need all this logic trying to figure out which video was clicked.